byway/src/main/webapp/destinations.js

```diff
@@ -180,11 +180,23 @@ function addLocationToDom(results,container){
 * fills Start location Searchbox with previously input
 */
 function updateStartDestination(locationData){
-    if (locationData.start == null){
+    if (locationData.start == "" || locationData.start == null){
       document.getElementById('start-search-box').value = "";
     }
     else{
-      document.getElementById('start-search-box').value = locationData.start;
+      const request = {
+        placeId: locationData.start,
+        fields: ["formatted_address"]
+      };
+      let service = new google.maps.places.PlacesService(map);
```

Comment:
Rather than declare multiple times, make into a global variable so it is made once and persists for all functions

Reply:
done